classes/controllers/grid/issues/IssueGridHandler.inc.php

```diff
@@ -298,25 +298,19 @@ function deleteIssue($args, $request) {
 		if (!$issue || !$request->checkCSRF()) return new JSONMessage(false);
 
 		$journal = $request->getJournal();
-		$isBackIssue = $issue->getPublished() > 0 ? true: false;
 
 		// remove all published submissions and return original articles to editing queue
-		import('classes.article.ArticleTombstoneManager');
-		$articleTombstoneManager = new ArticleTombstoneManager();
 		$submissionsIterator = Services::get('submission')->getMany([
 			'contextId' => $issue->getJournalId(),
-			'count' => 5000, // large upper-limit
 			'issueIds' => $issue->getId(),
 		]);
 		foreach ($submissionsIterator as $submission) {
-			$submission = Services::get('submission')->edit($submission, ['status' => STATUS_QUEUED], $request);
-			if ($isBackIssue) {
-				$articleTombstoneManager->insertArticleTombstone($submission, $journal);
-			}
 			$publications = (array) $submission->getData('publications');
 			foreach ($publications as $publication) {
-				$publication = Services::get('publication')->edit($publication, ['issueId' => ''], $request);
+				$publication = Services::get('publication')->edit($publication, ['issueId' => '', 'status' => STATUS_QUEUED], $request);
```

Comment:
Further below, in the TocGridHandler::removeArticle() we check if the publication has the same issue. Should we check it here too?

Reply:
Good catch! Fixed in https://github.com/pkp/ojs/commit/f2867376acdc2ace714908b9f66a86caebc94667